app/Wrappers/AzureApplicationInsight.php

```diff
@@ -0,0 +1,223 @@
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+namespace App\Wrappers;
+
+use ApplicationInsights\Telemetry_Client;
+use Exception;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use Illuminate\Queue\Events\JobProcessed;
+use Seat\Eveapi\Jobs\EsiBase;
+use Seat\Services\Exceptions\SettingException;
+
+/**
+ * Class AzureApplicationInsight.
+ * @package App\Wrappers
+ */
+class AzureApplicationInsight
+{
+    /**
+     * Determine the number of metrics which will put on the side before being sent to Azure Application Insight.
+     */
+    const MAXIMUM_QUEUE_SIZE = 10;
+
+    /**
+     * @var \ApplicationInsights\Telemetry_Client
+     */
+    private $telemetryClient;
+
+    /**
+     * AzureApplicationInsight constructor.
+     */
+    public function __construct()
+    {
+        $this->initContainer();
+    }
+
+    /**
+     * AzureApplicationInsight destructor.
+     */
+    public function __destruct()
+    {
+        // send all remaining queue elements to the telemetry platform
+        try {
+            $this->telemetryClient->flush();
+        } catch (RequestException $e) {
+            logger()->error($e->getMessage(), $e->getTrace());
+        }
+    }
+
+    /**
+     * Determine if yes or no the monitoring service is online.
+     *
+     * @return bool
+     */
+    public function isEnabled() : bool
+    {
+        // in case client is unset, monitoring is offline
+        if (! isset($this->telemetryClient))
+            return false;
+
+        // attempt to determine user choice related to monitoring
+        try {
+            $value = setting('allow_tracking', true);
+
+            return $value === 'yes';
+        } catch (SettingException $e) {
+            logger()->error($e->getMessage(), $e->getTrace());
+        }
+
+        // by default, assume monitoring is disabled
+        return false;
+    }
+
+    /**
+     * @param Exception $exception
+     */
+    public function trackException(Exception $exception)
+    {
+        // ensure telemetry service is active or attempt to init it
+        if (! $this->beforeTrackTelemetry())
+            return;
+
+        $this->telemetryClient->trackException($exception);
+
+        $this->afterTrackTelemetry();
+    }
+
+    /**
+     * @param $request
+     * @param $response
+     */
+    public function trackRequest($request, $response)
+    {
+        // ensure telemetry service is active or attempt to init it
+        if (! $this->beforeTrackTelemetry())
+            return;
+
+        // build a name composed of HTTP verb and request path
+        $name = sprintf('%s %s', strtoupper($request->method()), $request->path());
+
+        // build an extra properties array
+        $properties = [
+            'ajax' => $request->ajax(),
+            'ip' => $request->ip(),
```

Comment:
This must be anonymized: as of today and to be in line with the intend of Google Analytics and documentation about analytics

https://github.com/eveseat/services/blob/master/src/Jobs/Analytics.php#L173